controllers/nstemplateset/status.go

```diff
@@ -88,6 +90,50 @@ func (r *statusManager) setStatusReady(ctx context.Context, nsTmplSet *toolchain
 		})
 }
 
+// updateStatusClusterResourcesRevisions updates the cluster resources and features list in the status of the nstemplateset
+func (r *statusManager) updateStatusClusterResourcesRevisions(ctx context.Context, nsTmplSet *toolchainv1alpha1.NSTemplateSet) error {
+	updateFeatureAnnotation, featureAnnotation := featureAnnotationNeedsUpdate(nsTmplSet)
+	if updateFeatureAnnotation || clusterResourcesNeedsUpdate(nsTmplSet) {
+		// save the feature toggles into the status
+		// we updated also the featureToggles since the current the logic handles only cluster scoped resources.
+		// the logic could be refactored and transformed in something more generic, that can be reused for namespace scoped resources as well.
+		nsTmplSet.Status.FeatureToggles = featureAnnotation
+		nsTmplSet.Status.ClusterResources = nsTmplSet.Spec.ClusterResources
+		return r.Client.Status().Update(ctx, nsTmplSet)
+	}
+	return nil
+}
+
+// featureAnnotationNeedsUpdate checks if the feature annotation has changed on the nstemlpateset compared to what was last time saved in the status
+func featureAnnotationNeedsUpdate(nsTmplSet *toolchainv1alpha1.NSTemplateSet) (bool, []string) {
+	featureAnnotation := nsTmplSet.Annotations[toolchainv1alpha1.FeatureToggleNameAnnotationKey]
+	featureAnnotationList := utils.SplitCommaSeparatedList(featureAnnotation)
+	return !reflect.DeepEqual(featureAnnotationList, nsTmplSet.Status.FeatureToggles), featureAnnotationList
+}
+
+// clusterResourcesNeedsUpdate checks if there is a drift between the cluster resources set in the spec and the status of the nstemplateset
+func clusterResourcesNeedsUpdate(nsTmplSet *toolchainv1alpha1.NSTemplateSet) bool {
+	return (nsTmplSet.Status.ClusterResources == nil && nsTmplSet.Spec.ClusterResources != nil) ||
+		(nsTmplSet.Status.ClusterResources != nil && nsTmplSet.Spec.ClusterResources == nil) ||
+		nsTmplSet.Status.ClusterResources.TemplateRef != nsTmplSet.Spec.ClusterResources.TemplateRef
+}
+
+func (r *statusManager) updateStatusNamespacesRevisions(ctx context.Context, nsTmplSet *toolchainv1alpha1.NSTemplateSet) error {
+	if !reflect.DeepEqual(nsTmplSet.Spec.Namespaces, nsTmplSet.Status.Namespaces) {
```

Comment:
Is it safe to assume the same order of elements? As above, regardless of the answer, there are better alternatives to `reflect.DeepEqual()`.

Reply:
good point, see https://github.com/codeready-toolchain/member-operator/pull/648/commits/40b6fe52be26a984146b98a5b6ab43d36e85f9b0